main.go

```diff
@@ -140,6 +151,57 @@ func setupRouter() *gin.Engine {
 		c.JSON(http.StatusOK, result.Update.State == "capturing")
 	})
 
+	r.GET("/calendar", func(c *gin.Context) {
+		client := &http.Client{}
+		// Cutoff is set to 24 hours from now
+		cutoff := time.Now().UnixMilli() + int64(86400000)
+		url := config.Opencast.Url + "/recordings/calendar.json?agentid=" + config.Opencast.Agent + "&cutoff=" + fmt.Sprint(cutoff) + "&timestamp=true"
+		req, err := http.NewRequest("GET", url, nil)
+		req.SetBasicAuth(config.Opencast.Username, config.Opencast.Password)
+		resp, err := client.Do(req)
+		if err != nil {
+			log.Println(err)
+			c.JSON(http.StatusBadGateway, nil)
+			return
+		}
+		if resp.StatusCode != 200 {
+			log.Println(resp)
+			c.JSON(resp.StatusCode, nil)
+			return
+		}
+
+		bodyText, err := io.ReadAll(resp.Body)
+		s := string([]byte(bodyText))
+		start := regexp.MustCompile(`"startDate":[\d]+`)
+		end := regexp.MustCompile(`"endDate":[\d]+`)
+		t := regexp.MustCompile(`"event.title":"[^"]+"`)
+
+		startDates := start.FindAllString(s, -1)
+		endDates := end.FindAllString(s, -1)
+		titles := t.FindAllString(s, -1)
```

Comment:
Why do we use regular expressions to extract data from the JSON and not just parse the JSON?

Reply:
because the JSON contains some characters that aren't parsed correctly and that caused errors